crates/ruff/src/rules/flake8_bandit/rules/shell_injection.rs

```diff
@@ -0,0 +1,380 @@
+        vec!["os", "spawnlpe"],
+        vec!["os", "spawnv"],
+        vec!["os", "spawnve"],
+        vec!["os", "spawnvp"],
+        vec!["os", "spawnvpe"],
+        vec!["os", "startfile"],
+    ],
+});
+
+enum HasShell {
+    // The shell keyword argument is set and it evaluates to false
+    Falsey,
+    // The shell keyword argument is set and it evaluates to true
+    Truthy,
+    // The shell keyword argument is set but we cannot evaluate it
+    Unknown,
+}
+
+fn has_shell(keywords: &[Keyword]) -> Option<HasShell> {
+    if let Some(keyword) = find_shell_keyword(keywords) {
+        match &keyword.node.value.node {
+            ExprKind::Constant {
+                value: Constant::Bool(b),
+                ..
+            } => {
+                if *b {
+                    Some(HasShell::Truthy)
+                } else {
+                    Some(HasShell::Falsey)
+                }
+            }
+            ExprKind::Constant {
+                value: Constant::Int(int),
+                ..
+            } => {
+                if int == &BigInt::from(0u8) {
+                    Some(HasShell::Falsey)
+                } else {
+                    Some(HasShell::Truthy)
+                }
+            }
+            ExprKind::Constant {
+                value: Constant::Float(float),
+                ..
+            } => {
+                if (float - 0.0).abs() < f64::EPSILON {
+                    Some(HasShell::Falsey)
+                } else {
+                    Some(HasShell::Truthy)
+                }
+            }
+            ExprKind::List { elts, .. } => {
+                if elts.is_empty() {
+                    Some(HasShell::Falsey)
+                } else {
+                    Some(HasShell::Truthy)
+                }
+            }
+            ExprKind::Dict { keys, .. } => {
+                if keys.is_empty() {
+                    Some(HasShell::Falsey)
+                } else {
+                    Some(HasShell::Truthy)
+                }
+            }
+            ExprKind::Tuple { elts, .. } => {
+                if elts.is_empty() {
+                    Some(HasShell::Falsey)
+                } else {
+                    Some(HasShell::Truthy)
+                }
+            }
+            _ => Some(HasShell::Unknown),
+        }
+    } else {
+        None
+    }
+}
+
+fn find_shell_keyword(keywords: &[Keyword]) -> Option<&Keyword> {
+    keywords.iter().find(|keyword| {
+        keyword
+            .node
+            .arg
+            .as_ref()
+            .and_then(|arg| if arg == "shell" { Some(()) } else { None })
+            .is_some()
+    })
+}
+
+fn shell_call_looks_safe(arg: &Expr) -> bool {
+    matches!(
+        arg.node,
+        ExprKind::Constant {
+            value: Constant::Str(_),
+            ..
+        }
+    )
+}
+
+fn get_call_kind(checker: &mut Checker, func: &Expr) -> Option<CallKind> {
+    checker.ctx.resolve_call_path(func).and_then(|call_path| {
+        if CONFIG
```

Comment:
This looks like a great idea, if these will never change. In bandit these are configurable as you can see [here in the bandit docs](https://bandit.readthedocs.io/en/latest/config.html). I *think* `flake8-bandit` will make use of this configuration. If we want to support these being configurable in future it would be harder with this approach.

To address the performance concern I could flip the data structure to something like `HashMap<&str, CallKind>` and use the dotted module form as the key (eg `"os.system"`). I've implemented it a separate branch [over here](https://github.com/robyoung/ruff/blob/implement-bandit-shell-injection-rules-hashmap-config/crates/ruff/src/rules/flake8_bandit/rules/shell_injection.rs#L98) and it looks ok. Are there tools to help comparing the performance between the two approaches?

Reply:
My two cents: I think we should run with the match approach since it's super simple and satisfies our needs for now (and it lets us unblock + get this merged). If we want to extend these to be configurable, we can def continue exploring and benchmarking in a follow-up PR.

Looking at the Bandit docs, I _think_ that configuration is mostly used to turn rules on and off on a per-file or per-line basis, rather than to make (e.g.) the list of matching functions here configurable -- so it may not be needed anyway? But I could be misreading.

Reply:
Looking back at the docs, maybe it _is_ configurable? So I might be wrong on that. But we can still revisit if we opt to respect and implement that configuration. 

https://github.com/PyCQA/bandit/blob/main/bandit/plugins/injection_shell.py#L137

Reply:
Some examples of folks configuring this: https://cs.github.com/?q=%22shell_injection%22+language%3AYAML&p=1&pt=39cf6d9c09e5cd50a57cd21efcc19b97b3adf0be3156e284bd381526c71480c88c3689e17facd0388f6f735415b3e25423a39eb3425f52e3c01c1d3ede83c0856cf2ee13417f4e33c9f62672674d2694273a5a2987e885abd30da5d11bcfd6b7daf841dc69e2cea8b5c911767a46e9258d8a0792aef0dd660da846cb51b7c59a03c3fc0e30cd31d66374a2ddef50669d0b52857e08fa9bbddca9adb628bd9e505bdcd7c78d82cd9d7a568c7f7221e92be380d23ed1f561ec6599ee50f8f38465776e8b9918366148c7b57bbcd673de36a1a26468cddcdfb12e7b18715a436a456b5e662cd4e7dbcac99c2a11d41da69a89d2551dda249fce6f6be382cc3488752eff4ae2701ada51df2b290a515d6d20a92abfe2&scope=&scopeName=All+repos